source/Ship.cpp

```diff
@@ -704,24 +704,15 @@ void Ship::Save(DataWriter &out) const
 		out.Write("outfits");
 		out.BeginChild();
 		{
-			// Using a temporary map with the outfit names as keys so that
-			// outfits are written to the savefile in named alphabetical
-			// order.
-			// The game-engine doesn't care in which order outfits are
-			// saved or loaded, the sorting here is done as a service for
-			// content creators that use savegames for working on their
-			// new content.
-			map<string, int> orderedOutfits;
-			for(const auto &it : outfits)
-				if(it.first && it.second)
-					orderedOutfits[it.first->Name()] = it.second;
-			for(const auto &it : orderedOutfits)
-			{
-				if(it.second == 1)
-					out.Write(it.first);
-				else
-					out.Write(it.first, it.second);
-			}
+			using OutfitElement = pair<const Outfit *const, int>;
+			WriteSorted(outfits,
+				[](const OutfitElement *lhs, const OutfitElement *rhs){ return lhs->first->Name() < rhs->first->Name(); },
+				[&out](const pair<const Outfit *, int> &it){
```

Comment:
Is it possible to use OutfitElement here as well instead of the pair(Outfit, int>?